sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -556,19 +623,53 @@ private void throwIfAnalyzeStatusInvalid(SimpleResponse<AnalyzeOperationResult>
     private Function<PollingContext<OperationResult>, Mono<OperationResult>> analyzeFormStreamActivationOperation(
         Flux<ByteBuffer> data, String modelId, long length,
         FormContentType formContentType, boolean includeTextDetails) {
-        return (pollingContext) -> {
-            try {
-                Objects.requireNonNull(data, "'data' is required and cannot be null.");
-                Objects.requireNonNull(modelId, "'modelId' is required and cannot be null.");
-                Objects.requireNonNull(formContentType, "'formContentType' is required and cannot be null.");
+        Objects.requireNonNull(data, "'data' is required and cannot be null.");
+        Objects.requireNonNull(modelId, "'modelId' is required and cannot be null.");
+
+        // Size of buffered data, it used to collect all data from input Flux<ByteBuffer>
+        final long[] bufferedDataSize = {0};
+        // Cached byte buffer
+        final List<ByteBuffer> cachedBuffers = new LinkedList<>();
+        // A boolean value that only used once for guessing content type on input data
+        final AtomicBoolean[] isGuessed = {new AtomicBoolean(true)};
+        // Content type on input data
+        final ContentType[] contentType = new ContentType[1];
+
+        return pollingContext -> data.filter(ByteBuffer::hasRemaining)
```

Comment:
This logic appears to be duplicated three times, any chance this could be abstracted? Possibly a method that takes `Flux<ByteBuffer>` and returns `Mono<Tuple2<ContentType, Flux<ByteBuffer>>`.